### ISSUE TITLE
Noah-MP v4.0.1 Precipitation Logic Error when model time step is greater than forcing timestep 

### ISSUE THREAD

**Issue by TimLahmers** [NASA-LIS/LISF#710]
In rare situations where the Noah-MP v4.0.1 time step is greater (i.e. coarser) than the forcing timestep, the model receives half the precipitation. A symptom of this is the Rainf_tavg = (1/2)*Rainf_f_tavg in the LIS_HIST output files. 

@karsenau suggested a fix for this in May 2020 that would include removing in NoahMP401_main.F90:
tmp_prcp       = dt * (NOAHMP401_struc(n)%noahmp401(t)%prcp   / NOAHMP401_struc(n)%forc_count)

Then, this would be added in its place. This has been tested in LIS/WRF-Hydro:
tmp_dt         = NOAHMP401_struc(n)%ts
tmp_prcp       = tmp_dt * (NOAHMP401_struc(n)%noahmp401(t)%prcp   / NOAHMP401_struc(n)%forc_count)

Technically the following would be more correct:
if NOAHMP401_struc(n)%ts > LIS_rc%ts then
    tmp_dt         = NOAHMP401_struc(n)%ts
    tmp_prcp       = tmp_dt * (NOAHMP401_struc(n)%noahmp401(t)%prcp   / NOAHMP401_struc(n)%forc_count)
else
    tmp_prcp       = dt * (NOAHMP401_struc(n)%noahmp401(t)%prcp   / NOAHMP401_struc(n)%forc_count)
endif

@sujayvkumar @bmcandr could this be added to another PR like #618 @dmocko also helped suggest this solution. Have you tested this further?

**Comment by bmcandr** [comment]
Hi @TimLahmers,

Thanks for opening this issue. Rather than rolling this into #618, I recommend opening a PR specifically for this fix for two reasons:

1. PRs should be focused and as small as possible. This leads to quicker reviews.
2. This bug is present in the support/lisf-public-7.3 branch and should therefore be fixed there (i.e., branch off of support/lisf-public-7.3 to make the fix).

----

BTW here's how to format source code in GitHub comments:

* Inline code, such as `variables` or `function_names()`, can be surrounded with a single backtick (\`) on each end
* Multi-line code snippets can be "fenced" with lines containing triple backticks (\`\`\`). Optionally, syntax highlighting can be enabled by adding the language specifier after the opening backticks. The specifier can be the language name or file extension (e.g., python or f90). Here's how the proposed fix in your comment would be formatted:

<details>
<summary>Click for raw markdown</summary>

    ```f90
    if NOAHMP401_struc(n)%ts > LIS_rc%ts then
       tmp_dt = NOAHMP401_struc(n)%ts
       tmp_prcp = tmp_dt * (NOAHMP401_struc(n)%noahmp401(t)%prcp / NOAHMP401_struc(n)%forc_count)
    else
       tmp_prcp = dt * (NOAHMP401_struc(n)%noahmp401(t)%prcp / NOAHMP401_struc(n)%forc_count)
    endif
    ```
</details>

<details>
<summary>Click for rendered result</summary>

```f90
if NOAHMP401_struc(n)%ts > LIS_rc%ts then
   tmp_dt = NOAHMP401_struc(n)%ts
   tmp_prcp = tmp_dt * (NOAHMP401_struc(n)%noahmp401(t)%prcp / NOAHMP401_struc(n)%forc_count)
else
   tmp_prcp = dt * (NOAHMP401_struc(n)%noahmp401(t)%prcp / NOAHMP401_struc(n)%forc_count)
endif
```
</details>

**Comment by bmcandr** [comment]
Merged to master in 9eb0559.
